components/eamxx/src/share/util/scream_vertical_interpolation_impl.hpp

```diff
@@ -118,39 +52,248 @@ void perform_vertical_interpolation_impl_1d(
   team.team_barrier();
 }
 
+/* ---------------------------------------------------------------------- 
+ * Versions where x_tgt is a 2-D view
+ * ---------------------------------------------------------------------- */
 template<typename T, int P, int N> 
-void perform_vertical_interpolation_impl_Nd(
-  const view_Nd<const Pack<T,P>,2>& x_src,
+void perform_checks(
+  const view_2d<const Pack<T,P>>&   x_src,
+  const view_2d<const Pack<T,P>>&   x_tgt,
+  const view_Nd<const Pack<T,P>,N>& input,
+  const view_Nd<      Pack<T,P>,N>& output,
+  const int                         nlevs_src,
+  const int                         nlevs_tgt)
+{
+  auto rank = input.rank;
+  EKAT_REQUIRE_MSG (rank>1 &&rank<=4,"Error::scream_vertical_interpolation, passed view of rank (" + std::to_string(rank) +"), only support ranks 2-4\n");
+
+  // The input data and x_src data should match in the appropriate size
+  EKAT_REQUIRE(x_src.extent_int(0) == input.extent_int(0));
+  EKAT_REQUIRE(x_src.extent_int(1) == input.extent_int(input.rank-1));
+  // The output data and x_tgt data should match in the appropriate size
+  EKAT_REQUIRE(x_tgt.extent_int(0) == output.extent_int(0));
+  EKAT_REQUIRE(x_tgt.extent_int(1) == output.extent_int(input.rank-1));
+
+
+  // The output and input data should match in rank
+  EKAT_REQUIRE(input.rank == output.rank);
```

Comment:
This check is poinltess. The two template args are already view_Nd with the same N. There's no way the compiler would pick this template signature if input/output had different ranks.